File: mobile/lib/data/models/diet/food.dart
class Food {
  final int id;
  final String name;
  final int calories;
  final double protein;
  final double fat;
  final double carbohydrates;

  Food({
    required this.id,
    required this.name,
    required this.calories,
    required this.protein,
    required this.fat,
    required this.carbohydrates,
  });

  factory Food.fromJson(Map<String, dynamic> json) {
    return Food(
      id: json['id'],
      name: json['name'],
      calories: json['calories'],
      protein: json['protein'].toDouble(),
      fat: json['fat'].toDouble(),
      carbohydrates: json['carbohydrates'].toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'calories': calories,
      'protein': protein,
      'fat': fat,
      'carbohydrates': carbohydrates,
    };
  }
}

File: backend/app/Models/Food.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

/**
 * @method static when(array|string|null $search, \Closure $param)
 */
class Food extends Model
{
    use HasFactory;

    protected $table = 'food';

    protected $fillable = [
        'name', 'calories', 'protein', 'fat', 'carbohydrates'
    ];

    protected $casts = [
        'calories'      => 'integer',
        'protein'       => 'float',
        'fat'           => 'float',
        'carbohydrates' => 'float',
    ];

    public function dietEntries(): HasMany
    {
        return $this->hasMany(DietEntry::class);
    }

    public function scopeSearch($query, $search): void
    {
        if ($search) {
            $query->where('name', 'like', "%{$search}%");
        }
    }
}
